File: myskills/backend/app/routers/vector.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.database import get_db
from app.services.vector import vector_service
from typing import List

router = APIRouter()

@router.get("/knowledge-bases/{kb_id}/vectors", response_model=List[dict])
async def get_vectors(kb_id: str, db: Session = Depends(get_db)):
    """获取向量列表"""
    try:
        return vector_service.get_vectors(db, kb_id)
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

@router.post("/knowledge-bases/{kb_id}/vectors/rebuild", response_model=dict)
async def rebuild_vectors(kb_id: str, db: Session = Depends(get_db)):
    """重建向量索引"""
    try:
        return vector_service.rebuild_vectors(db, kb_id)
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

@router.post("/retrieve", response_model=List[dict])
async def retrieve_vectors(query_data: dict, db: Session = Depends(get_db)):
    """向量检索"""
    try:
        query = query_data.get("query")
        knowledge_base_id = query_data.get("knowledge_base_id")
        if not query:
            raise ValueError("查询文本不能为空")
        return vector_service.retrieve_vectors(db, query, knowledge_base_id)
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

File: myskills/frontend/src/services/vector.ts
import api from './api';

interface Vector {
  id: string;
  document_id: string;
  chunk_id: string;
  content: string;
  embedding_dimension: number;
  created_at: string;
}

interface RetrieveResult {
  id: string;
  content: string;
  similarity: number;
  document_id: string;
}

class VectorService {
  async getVectors(kbId: string): Promise<Vector[]> {
    return api.get(`/knowledge-bases/${kbId}/vectors`);
  }

  async rebuildVectors(kbId: string): Promise<{ message: string }> {
    return api.post(`/knowledge-bases/${kbId}/vectors/rebuild`);
  }

  async retrieveVectors(query: string, kbId?: string): Promise<RetrieveResult[]> {
    return api.post('/vectors/retrieve', { query, knowledge_base_id: kbId });
  }
}

export default new VectorService();
